### PR TITLE
chore: TRACEFOSS-2725 support for investigation join table filtering

### PR DESCRIPTION
**Pull request by ds-ext-sceronik**

### CODE REVIEW

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/repository/BaseSpecification.java [hunk 1]
```diff
@@ -20,12 +20,16 @@
...
+import jakarta.persistence.criteria.Join;
+import jakarta.persistence.criteria.Path;
 import jakarta.persistence.criteria.Predicate;
 import jakarta.persistence.criteria.Root;
 import lombok.Getter;
+import org.eclipse.tractusx.traceability.common.domain.ParseLocalDateException;
 import org.eclipse.tractusx.traceability.common.model.SearchCriteriaFilter;
 import org.eclipse.tractusx.traceability.common.model.SearchCriteriaOperator;
 import org.eclipse.tractusx.traceability.common.model.SearchCriteriaStrategy;
+import org.jetbrains.annotations.NotNull;
```
Comment (ds-lcapellino) [inline tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/repository/BaseSpecification.java:32]: Do we generally use jetbrains annotations? There is already a try/catch in the used method, so maybe this is obsolete?

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/repository/BaseSpecification.java [hunk 2]
```diff
@@ -50,37 +54,75 @@ protected BaseSpecification(SearchCriteriaFilter searchCriteriaFilter) {
...
+    private Path<Object> getFieldPath(Root<?> root, SearchCriteriaFilter criteria) {
+        if (isJoinQueryFieldName(criteria.getKey())) {
+            Join<?, ?> join = root.join(getJoinTableName(criteria.getKey()));
+            return join.get(getJoinTableFieldName(criteria.getKey()));
+        }
+        return root.get(criteria.getKey());
+    }
+
+    @NotNull
+    private static LocalDate getParseLocalDate(String fieldValue, String fieldName) {
```
Comment (ds-lcapellino) [inline tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/repository/BaseSpecification.java:104]:
```suggestion
    private static LocalDate from(String fieldValue, String fieldName) {
```

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/repository/BaseSpecification.java [hunk 3]
```diff
@@ -50,37 +54,75 @@ protected BaseSpecification(SearchCriteriaFilter searchCriteriaFilter) {
...
-            return builder.greaterThanOrEqualTo(root.get(criteria.getKey()),
+            final LocalDate localDate = getParseLocalDate(expectedFieldValue, fieldName);
+
+            return builder.greaterThanOrEqualTo(fieldPath.as(LocalDateTime.class),
                     LocalDateTime.of(localDate, LocalTime.MIN));
         }
         return null;
     }
 
+    private Path<Object> getFieldPath(Root<?> root, SearchCriteriaFilter criteria) {
```
Comment (ds-lcapellino) [inline tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/repository/BaseSpecification.java:95]:
```suggestion
    private Path<Object> from(Root<?> root, SearchCriteriaFilter criteria) {
```

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/repository/BaseSpecification.java (hunk 2 above)
Comment (ds-ext-sceronik) [reply tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/repository/BaseSpecification.java:104]: As usage of this in the above method goes It is more clear to read when method indicate what it returns

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/repository/BaseSpecification.java (hunk 3 above)
Comment (ds-ext-sceronik) [reply tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/repository/BaseSpecification.java:95]: as above . more clear what is happening withing specification predicate building method when method indicates what it returns. in coming sprints there might be joinColumnFieltPaths used and other type of paths so I would keep te name

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/domain/ParseLocalDateException.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+package org.eclipse.tractusx.traceability.common.domain;
+
+public class ParseLocalDateException extends RuntimeException {
+    public ParseLocalDateException(String localDate, String fieldName) {
+        super("Could not parse '%s' for field '%s'. Accepted LocalDate format is following 'YYYY-MM-DD'".formatted(localDate, fieldName));
```
Comment (ds-lcapellino) [inline tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/domain/ParseLocalDateException.java:24]:
```suggestion
    public ParseLocalDateException(String localDate, String fieldName, Throwable cause) {
        super("Could not parse '%s' for field '%s'. Accepted LocalDate format is following 'YYYY-MM-DD'".formatted(localDate, fieldName), cause);
```

**Review by ds-lcapellino — APPROVED** [review]
LGTM

**Pull request merged**